portal-bridge/src/bridge.rs

```diff
@@ -79,9 +79,18 @@ impl Bridge {
     }
 
     async fn launch_test(&self, test_path: PathBuf) {
-        let test_asset = fs::read_to_string(test_path).expect("Error reading test asset.");
-        let assets: TestAssets =
-            serde_json::from_str(&test_asset).expect("Unable to parse test asset.");
+        let test_asset = fs::read_to_string(test_path.clone()).expect("Error reading test asset.");
+        let extension = test_path
+            .extension()
+            .and_then(|x| x.to_str())
+            .expect("Unable to parse test path extension");
```

Comment:
I believe the `.clone()` can be avoided by flipping the order of these two. The reason I think so is because `PathBuff::extension()` just takes a reference to the path, but the `read_to_string()` consumes the object.

```suggestion
        let extension = test_path
            .extension()
            .and_then(|x| x.to_str())
            .expect("Unable to parse test path extension");
        let test_asset = fs::read_to_string(test_path).expect("Error reading test asset.");
```

Reply:
```rust
error[E0505]: cannot move out of `test_path` because it is borrowed
  --> portal-bridge/src/bridge.rs:87:45
   |
81 |       async fn launch_test(&self, test_path: PathBuf) {
   |                                   --------- binding `test_path` declared here
82 |
83 |           let extension = test_path
   |  _________________________-
84 | |             .extension()
   | |________________________- borrow of `test_path` occurs here
...
87 |           let test_asset = fs::read_to_string(test_path).expect("Error reading test asset.");
   |                                               ^^^^^^^^^ move out of `test_path` occurs here
88 |
89 |           let assets: TestAssets = match extension {
   |                                          --------- borrow later used here

For more information about this error, try `rustc --explain E0505`.
```

I could do this if you want
```rust
        let extension = test_path
            .extension()
            .and_then(|x| x.to_str())
            .expect("Unable to parse test path extension").to_string();
        let test_asset = fs::read_to_string(test_path).expect("Error reading test asset.");

        let assets: TestAssets = match &extension[..] {
            "json" => serde_json::from_str(&test_asset).expect("Unable to parse json test asset."),
            "yaml" => serde_yaml::from_str(&test_asset).expect("Unable to parse yaml test asset."),
            _ => panic!("Invalid file extension"),
        };
        ```

Reply:
I think I can use 2 expects instead of the and then I will try that

Reply:
Nvm I just tried the idea I said and it still says a move happened